terraform/azure/modules/aks/aks.tf

```diff
@@ -0,0 +1,120 @@
+/*
+Not required for now. Saving this for later use. Review and delete this later
+
+# Generate random resource group name
+#resource "random_pet" "rg_name" {
+#  prefix = var.resource_group_name_prefix
+#}
+
+#resource "random_pet" "azurerm_kubernetes_cluster_name" {
+#  prefix = "cluster"
+#}
+
+#resource "random_pet" "azurerm_kubernetes_cluster_dns_prefix" {
+#  prefix = "dns"
+#}
+*/
+
+data "azuread_service_principal" "akssp"{
+    display_name =  var.azuread_service_principal_display_name
+}
+
+data "azurerm_resource_group" "rg" {
+    name     = var.modern_resource_group_name 
+}
+
+resource "azurerm_kubernetes_cluster" "aks" {
```

Comment:
I thought we were going to use the terraform module for the cluster creation? This is so we do not have to manage all the individual pieces. Just like we do with eks.
https://registry.terraform.io/modules/Azure/aks/azurerm/latest?tab=resources

Reply:
This is part of Azurerm.